Dockerfile

```diff
@@ -0,0 +1,28 @@
+FROM mhart/alpine-node:5.0
+
+ENV BUILD_PACKAGES bash curl-dev ruby-dev build-base libffi-dev libpq postgresql-dev git
+ENV RUBY_PACKAGES ruby ruby-io-console ruby-bundler ruby-raindrops ruby-nokogiri
+
+# Update and install all of the required packages.
+# At the end, remove the apk cache
+RUN apk update && \
+    apk upgrade && \
+    apk add $BUILD_PACKAGES && \
+    apk add $RUBY_PACKAGES && \
+    rm -rf /var/cache/apk/*
+
+RUN mkdir /app
+WORKDIR /app
+
+ADD . /app
+RUN bundle install --jobs 4 --without development test
```

Comment:
The main difference of not copying the `Gemfile` and `Gemfile.lock` before (I guess in this case, you'd also have to copy the `gemspec` file), is that the build will have to bundle all gems every time there's any change in the `app/`'s files.

If we just `COPY` the files bundle needs to know about first, then the bundling phase will only happen if there are changes in the file. Else, it'll use the cached layer which is all the gems installed.